### PR TITLE
improve mouse pointer in piano roll editor

### PR DESCRIPTION
**Pull request by BaraMGB**
I can't make screenshots with mouse pointer. 

This PR change the following things:
- the mouse pointer changes its appearance proper on all borders of the key area
- we use a pencil ( or eraser ) instead of the arrow with pencil icon
- the rubberband icon don't stuck anymore on a border when the mouse pointer leaves the area

### CODE REVIEW

File: src/gui/editors/PianoRoll.cpp [hunk 1]
```diff
@@ -216,7 +218,7 @@ PianoRoll::PianoRoll() :
 		m_noteEditMenu->addAction( act );
 	}
 	connect( signalMapper, SIGNAL(mapped(int)),
-			this, SLOT(changeNoteEditMode(int)) );
+			 this, SLOT(changeNoteEditMode(int)) );
```
Comment (Umcaruje) [inline src/gui/editors/PianoRoll.cpp]: Why is a space added here

File: src/gui/editors/PianoRoll.cpp [hunk 2]
```diff
@@ -261,32 +263,32 @@ PianoRoll::PianoRoll() :
 	if( s_whiteKeySmallPm == NULL )
 	{
 		s_whiteKeySmallPm = new QPixmap( embed::getIconPixmap(
-						"pr_white_key_small" ) );
+											 "pr_white_key_small" ) );
```
Comment (Umcaruje) [inline src/gui/editors/PianoRoll.cpp]: Why are all these lines more indented

File: src/gui/editors/PianoRoll.cpp [hunk 3]
```diff
@@ -319,21 +321,21 @@ PianoRoll::PianoRoll() :
 
 	// add time-line
 	m_timeLine = new TimeLineWidget( WHITE_KEY_WIDTH, 0, m_ppt,
-					Engine::getSong()->getPlayPos(
-						Song::Mode_PlayPattern ),
-						m_currentPosition, this );
+									 Engine::getSong()->getPlayPos(
+										 Song::Mode_PlayPattern ),
```
Comment (Umcaruje) [inline src/gui/editors/PianoRoll.cpp]: and here

**Comment by Umcaruje** [comment]
I'll now go and test this out

**Comment by Umcaruje** [comment]
This works nice, and feels a lot more natural. However, the automation editor is now inconsistent with this, so I guess the same thing should be done there. Other than the indentation issues, this looks ok to me.

**Comment by BaraMGB** [comment]
Fine, I'll fix it. Perhaps this should be tested on Apple and Windows, too.

File: src/gui/editors/PianoRoll.cpp [hunk 4]
```diff
@@ -1980,27 +1985,40 @@ void PianoRoll::mouseMoveEvent( QMouseEvent * me )
...
 					m_oldNotesEditHeight - ( me->y() - m_moveStartY ),
 					NOTE_EDIT_MIN_HEIGHT,
 					height() - PR_TOP_MARGIN - NOTE_EDIT_RESIZE_BAR -
-									PR_BOTTOM_MARGIN - KEY_AREA_MIN_HEIGHT );
+					PR_BOTTOM_MARGIN - KEY_AREA_MIN_HEIGHT );
 		repaint();
 		return;
 	}
-
+
```
Comment (Umcaruje) [inline src/gui/editors/PianoRoll.cpp]: unecessary   white space here

**Comment by BaraMGB** [comment]
@tresf can you test this on mac, please?

**Comment by tresf** [comment]
> I can't make screenshots with mouse pointer.

You can with a cellphone or a virtual machine.

> @tresf can you test this on mac, please?

I don't see any OS-specific code, so I'm not sure why this is needed.  I have MacOS Qt4 and Qt5 builds here: https://github.com/tresf/lmms/releases/tag/v1.2.0-RC2-MOUSE

Test results:
* [x] the mouse pointer changes its appearance proper on all borders of the key area
  * Yes, the piano keys seem to be ok on hover although the cursor [lingers after hovering over a note](https://cloud.githubusercontent.com/assets/6345473/19220980/d5c7f288-8e07-11e6-98a1-ea359a0d0a99.gif).
* [ ] we use a pencil ( or eraser ) instead of the arrow with pencil icon
  * No, the 4-directional arrow is what's displayed on Mac.
* [x] the rubberband icon don't stuck anymore on a border when the mouse pointer leaves the area
  * This seems to be true although the [cursor/marker area seems to still have the lasso](https://cloud.githubusercontent.com/assets/6345473/19221000/607cdf1a-8e08-11e6-81ba-4e627265030c.gif), I'd assume it would disappear there, not the toolbar.

**Comment by BaraMGB** [comment]
>I don't see any OS-specific code, so I'm not sure why this is needed.

Because the behavior is completely different from what I see on Linux. 😬

**Comment by liushuyu** [comment]
So is this cooled down or still working in progress? I think this fix is somewhat important

**Comment by BaraMGB** [comment]
I need A video of what's happening in macOS. @simonvanderveldt?

**Comment by BaraMGB** [comment]
I would like to merge this into master, but I can not test if this works on MacOS. Can someone test this?

**Comment by tresf** [comment]
> I would like to merge this into master, but I can not test if this works on MacOS. Can someone test this?

Test what?  I don't see any new commits since my testing on 2016-10-09

**Comment by zonkmachine** [comment]
This works well over here.

> - No, the 4-directional arrow is what's displayed on Mac.

I had a peek at the qt bug tracker but nothing popped up as an absolute fit for this case. Though there were other reports of wrong cursors displayed.


@BaraMGB If this needs more testing you might wanna squash and rebase it.

**Comment by BaraMGB** [comment]
>although the cursor lingers after hovering over a note.

@tresf  can you test please if the Qt::SizeAllCursor is shown anymore and if it's stuck on leaving the note?

**Comment by zonkmachine** [comment]
I just noticed the cursor is a bit nervous around the border to the NoteEditArea. It's a bit difficult to hit the spot where you resize and when you're doing the ActionResizeNoteEditArea the cursor is flickering.

**Comment by tresf** [comment]
> can you test please if the Qt::SizeAllCursor is shown anymore and if it's stuck on leaving the note?

Just re-tested on fresh `BaraMGB/pianorollmousepointer` branch and it's still stuck on Mac.  Results are identical to the animated GIFs from https://github.com/LMMS/lmms/pull/3065#issuecomment-252488464.

**Comment by BaraMGB** [comment]
I'll close this out. At the moment I have no chance to debug it on a mac. Perhaps I'll find a way in the future.

**Comment by zonkmachine** [comment]
Maybe a hint in https://github.com/LMMS/lmms/issues/3926

**Pull request closed without merging**